nm: use interface name instead of UUID

cathay4t opened the pull request:
Current nmstate code is using UUID when referring the controller/parent in
NM profile, this cause problem when the controller/parent is in-memory
profile which change UUID after reboot. The OpenShift is using in-memory
profile for `br-ex` and will recreate it after reboot. This prevent us from
modifying the ports of `br-ex`.

Using interface name for controller/parent reference would fix this
problem.

Resolves: https://issues.redhat.com/browse/RHEL-26270

Existing test cases are enough to test this change.

cathay4t commented:
Waiting NetworkManager fix: https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/merge_requests/1878

ihuguet commented on rust/src/lib/nm/query_apply/apply.rs:
```diff
@@ -445,3 +459,35 @@ fn check_nm_version(nm_api: &NmApi) {
+        if let Some(ctrl_name) = nm_conn.controller() {
+            if default_pvid_changed_brs.contains(&ctrl_name) {
+                log::info!(
+                    "Deactivating linux bridge port as its controller \
+                    has `vlan-default-pvid` changes"
```
We should mention here the name of the port being deactivated. Also, the message can be a bit confusing, a user might think that it's going to be kept deactivated. Maybe change to something like "Deactivating bridge port {bridge_port} to apply changes in the controller's 'vlan-default-pvid'".

cathay4t replied:
Changed to `Reactivating`

ihuguet commented on rust/src/lib/query_apply/bond.rs:
```diff
@@ -41,3 +45,63 @@ impl BondOptions {
 }
+
+impl MergedInterface {
+    pub(crate) fn get_bond_queue_id_changed_ports(&self) -> Vec<&str> {
```
Can you rename to `get_bond_ports_with_queue_id_changed`? I think it's more clear.

ihuguet commented on rust/src/lib/nm/query_apply/apply.rs:
```diff
@@ -491,3 +498,34 @@ fn is_bridge_port_changed_default_pvid(
+        if let Some(iface_name) = nm_conn.iface_name() {
+            if changed_ports.contains(&iface_name) {
+                log::info!(
+                    "Deactivating bond port {iface_name} as its \
+                    queue ID has changed"
```
As in the other commit, not sure if the message is totally clear. Maybe something like "Deactivating bond port {iface_name} to apply queue ID change".

cathay4t replied:
Changed to `Reactivating`.

ihuguet reviewed (commented):
The PR description doesn't say anything about bridge default-pvid and bond ports queue id.

cathay4t committed:
nm linux bridge: Refresh port vlan if `vlan-default-pvid` changed

Due to bug of NetworkManager:

    https://issues.redhat.com/browse/RHEL-26750

Reapply on linux bridge port will not refresh `vlan-default-pvid`
changes on linux bridge, hence we need to deactivate the bridge port
before reapply.

The existing test case
`test_linux_bridge_option_vlan_default_pvid_no_filtering` has this use
case covered.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t committed:
bond: Check overlap on bond port queue id

In kernel code drivers/net/bonding/bond_options.c
bond_option_queue_id_set(), kernel is not allowing multiple bond port
holding the same queue ID, hence we raise error when queue id overlapped.

When changing bond port queue_id, it might also overlap with previous
set bond port, hence we deactivate before activate bond port if its
queue_id changed.

Unit test case included for queue id overlap.

Existing Integration test case
`test_add_bond_with_port_config_and_modify` covered the desired queue_id
overlap with current queue_id.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t committed:
nm: use interface name instead of UUID

Current nmstate code is using UUID when referring the controller/parent in
NM profile, this cause problem when the controller/parent is in-memory
profile which change UUID after reboot. The OpenShift is using in-memory
profile for `br-ex` and will recreate it after reboot. This prevent us
from modifying the ports of `br-ex`.

Using interface name for controller/parent reference would fix this problem.

Existing test cases are enough to test this change.

Signed-off-by: Gris Ge <fge@redhat.com>
